### PR TITLE
Support IAM role

### PR DESCRIPTION
**Pull request by TelmoSenseFly**
We want boto to be able to assume role provider when credentials are not passed as parameters

https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#iam-role

### CODE REVIEW

File: attachment_s3/models/ir_attachment.py [hunk 1]
```diff
@@ -51,10 +51,11 @@ def _get_s3_bucket(self, name=None):
         secret_key = os.environ.get('AWS_SECRET_ACCESS_KEY')
         bucket_name = name or os.environ.get('AWS_BUCKETNAME')
 
-        params = {
```
Comment (simahawk) [inline attachment_s3/models/ir_attachment.py:54]: `params` would be undefined -> broken

File: attachment_s3/models/ir_attachment.py [hunk 2]
```diff
@@ -63,19 +64,6 @@ def _get_s3_bucket(self, name=None):
             params['region_name'] = region_name
         else:
             connect_s3 = boto.connect_s3
-        if not (access_key and secret_key and bucket_name):
```
Comment (simahawk) [inline attachment_s3/models/ir_attachment.py:66]: you cannot trash this so easily :)
Options:
1. override `_get_s3_bucket` in your own code. Easy and fast. No clash w/ us.
2. if you think this is something worth to support generically, you could add an extra env var like `AWS_IAM` or similar. Then if this var is present you bypass this part.

**Review by simahawk — CHANGES_REQUESTED** [review]

File: attachment_s3/models/ir_attachment.py (hunk 2 above)
Comment (TelmoSenseFly) [reply attachment_s3/models/ir_attachment.py:66]: Option 1, I will override ```_get_s3_bucket``` ;)

**Comment by TelmoSenseFly** [comment]
@simahawk

**Pull request closed without merging**